onnxruntime/contrib_ops/cuda/bert/attention_impl.cu

```diff
@@ -858,6 +858,10 @@ Status QkvToContext(
       query = data.query;
     }
 
+    DUMP_TENSOR_D("attention q(BSNH)", q, batch_size * sequence_length, num_heads * qk_head_size);
+    DUMP_TENSOR_D("attention k(BSNH)", k, batch_size * sequence_length, num_heads * qk_head_size);
+    DUMP_TENSOR_D("attention v(BSNH)", v, batch_size * sequence_length, num_heads * v_head_size);
+
     MemoryEfficientAttentionParams p;
     p.sm = device_prop.major * 10 + device_prop.minor;
     p.is_half = sizeof(T) == 2;
```

Comment:
Instead of 2, would it be clearer to say `sizeof(half)`? Or `std::is_same_v<T, half>`?